rust/extensions.bzl

```diff
@@ -0,0 +1,42 @@
+"Module extensions for using rules_rust with bzlmod"
+
+load(
+    "//rust/private:repository_utils.bzl",
+    "DEFAULT_EXTRA_TARGET_TRIPLES",
+    "DEFAULT_NIGHTLY_VERSION",
+    "DEFAULT_STATIC_RUST_URL_TEMPLATES",
+)
+load(":repositories.bzl", "rust_register_toolchains")
+
+def _rust_impl(ctx):
+    mod = ctx.modules[0]
```

Comment:
Nit: Don't think this is guaranteed to be the root module.

`mod = [m for m in ctx.modules if m.is_root][0]`